builder/qemu/step_run.go

```diff
@@ -16,41 +16,45 @@ import (
 // stepRun runs the virtual machine
 type stepRun struct {
 	DiskImage bool
-}
 
-type qemuArgsTemplateData struct {
-	HTTPIP      string
-	HTTPPort    int
-	HTTPDir     string
-	OutputDir   string
-	Name        string
-	SSHHostPort int
+	atLeastVersion2 bool
+	ui              packer.Ui
 }
 
 func (s *stepRun) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	config := state.Get("config").(*Config)
 	driver := state.Get("driver").(Driver)
-	ui := state.Get("ui").(packer.Ui)
+	s.ui = state.Get("ui").(packer.Ui)
 
-	// Run command is different depending whether we're booting from an
-	// installation CD or a pre-baked image
-	bootDrive := "once=d"
-	message := "Starting VM, booting from CD-ROM"
-	if s.DiskImage {
-		bootDrive = "c"
-		message = "Starting VM, booting disk image"
+	// Figure out version of qemu; store on step for later use
+	rawVersion, err := driver.Version()
```

Comment:
this information is needed later when defining device args, but it's easier to handle and put on the step before the defaulting so we can refer to the boolean when we need it.